internal/run.go

```diff
@@ -132,11 +136,22 @@ func (proc *cmdProcess) Start() error {
 	return proc.cmd.Start()
 }
 
-func (proc *cmdProcess) Kill() error {
+func (proc *cmdProcess) Stop() error {
 	if proc.cmd.Process == nil {
 		return nil
 	}
-	return proc.cmd.Process.Kill()
+
+	if runtime.GOOS == "windows" {
+		return proc.cmd.Process.Kill()
+	}
+
+	go func() {
+		// TODO: Make the wait time configurable.
+		time.Sleep(maxProcStopWait)
+		_ = syscall.Kill(-proc.cmd.Process.Pid, syscall.SIGKILL)
+	}()
+
+	return syscall.Kill(-proc.cmd.Process.Pid, syscall.SIGTERM)
```

Comment:
This logic is strange to me. We async start a kill after a sleep, then we attempt a term and return that error code. Then the calling code needs the `waitDone` logic b/c this Stop function doesn't actually wait for the kill. If instead, this function is Kill(gracePeriod), and you do the term first, then race between proc.cmd.Wait() and time.Sleep(), now the call site can just be `Kill(gracePeriod)` and the `waitDone` thing is not necessary. Right?